plugin/src/main/java/org/opensearch/ml/action/profile/MLProfileRequest.java

```diff
@@ -6,22 +6,28 @@
 package org.opensearch.ml.action.profile;
 
 import java.io.IOException;
+import java.util.Set;
 
 import org.opensearch.action.support.nodes.BaseNodesRequest;
 import org.opensearch.core.common.io.stream.StreamInput;
 import org.opensearch.core.common.io.stream.StreamOutput;
 import org.opensearch.ml.profile.MLProfileInput;
 
 import lombok.Getter;
+import lombok.Setter;
 
 public class MLProfileRequest extends BaseNodesRequest<MLProfileRequest> {
 
     @Getter
     private MLProfileInput mlProfileInput;
+    @Getter
+    @Setter
+    private Set<String> hiddenModelIds;
 
     public MLProfileRequest(StreamInput input) throws IOException {
         super(input);
         mlProfileInput = new MLProfileInput(input);
+        hiddenModelIds = input.readSet(StreamInput::readString);
```

Comment:
Add BWC version checking? Profile request will broadcast to all nodes, if the cluster has two versions, it may throw exception.